includes/Core/Modules/Module.php

```diff
@@ -283,11 +301,11 @@ public function is_connected() {
 				continue;
 			}
 
-			$key       = ! empty( $dataset->key ) ? $dataset->key : wp_rand();
-			$datapoint = $dataset->datapoint;
-			$data      = ! empty( $dataset->data ) ? (array) $dataset->data : array();
+			$key                   = $dataset->key ?: wp_rand();
```

Comment:
@felixarntz Should we not calculate the proper md5 for the request here if the key is not provided?

Reply:
Let's not worry about this now, but it's a good point for an enhancement. For our own purposes the current way works because we always pass a key. But generally I agree we should improve this. Either make passing a key required, or determine the fallback key the same way as JS.